base/ltlists.dtx

```diff
@@ -715,7 +715,65 @@
 % \cs{everypar}.
 %
 % This allows paragraph-making environments to work right when called
-% by other environments. (Changed 27 Oct 86)
+% by other environments. (Changed 27 Oct 86).
+%
+% In 2024 this logic was partially replaced with a new algorithm:
+% \begin{itemize}
+% \item
+%   \cs{if@endpe} is now set globally to \texttt{true} or \texttt{false}.
+% \item
+%    In addition \cs{@endpetrue} also initiates an \cs{aftergroup} call
+%    to \cs{propagate@doendpe} if it is called inside a group.
+% \item
+%    \cs{propagate@doendpe} in turn checks the status of \cs{if@endpe}
+%    and if that is \texttt{true} it calls \cs{@doendpe} otherwise it
+%    does nothing.
+% \item
+%    \cs{@doendpe} in turn calls \cs{@endpetrue} and also changes
+%    \cs{par} and \cs{everypar} to do end environment \cs{par}
+%    handling as before.
+% \item
+%    Because of \cs{@endpetrue} we get another \cs{aftergroup} so the
+%    mechanisnm slowly migrates out of several groups if those follow
+%    immediately after the end of the environment. If however, there is a new
+%    paragraph started or an explicit \cs{par} before the next group ends then this will
+%    result in a call to \cs{@endpefalse} and the migration stops
+%    (\cs{propagate@doendpe} is still called once after the group but does nothing).
+% \item
+%    Using this approach something like
+%\begin{verbatim}
+% {%  some customization here
+%  \begin{equation}
+%    x=y
+%  \end{equation}}
+% some text
+%\end{verbatim}
+%    is still correctly identified as a paragraph continuation so that
+%    there is no indentation before \texttt{some text}.
+% \item
+%    We can get away with using global settings of \cs{if@endpe} even
+%    in nested situations (without keeping track of the status in a
+%    stack), because the switch change is made only at
+%    the very end of such environments, basically directly before the
+%    \cs{endgroup} in \cs{end} and it is set back to \texttt{false} on
+%    the next \cs{everypar} or the next \cs{par}. Even if the
+%    environment is called without using \cs{begin} \cs{end} the
+%    situation doesn't change (or rather causes a problem).
+% \item
+%    However, there is one scenario where the new approach would
+%    change the behavior. If a box is being built, e.g., with
+%    \cs{setbox}, then we have now the case that a \cs{@endpetrue}
+%    inside would migrate out into a context in which it should not be
+%    true (because the box might get used elsewhere, e.g., a float).
+%    In the past, due to local switch changes, that didn't happen, i.e.,
+%    it would revert to \texttt{false} at the end of the box
+%    definition.
+% \item
+%    Thus, to avoid that one has to explicitly set it back to false at
```

Comment:
And: not clear what "it" is here.

Reply:
But Germans are so much more perceptive than I? 😃 

"it" was e en in a previous item!

Reply:
no, not more perceptive, but more tuned to keeping track of various "its" and "these" in sentences because our grammar often puts verbs far away so one has to keep much more tacking info mentally. More perceptive to both of us seems to be Jamanese language leaving out the he/she/it/me/you and leave you guessing which is meant.